### PR TITLE
Assessment runner refactors

### PR DESCRIPTION
**Pull request by stevenhorsman**
Various refactors to the assessment helper and assessment runner to try and abstract out complexity of the large runner Assess section, to make it easier to read and reduce duplication

- Drive by fix to drop the rust version post the merge of #2121

### CODE REVIEW

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go [hunk 1]
```diff
@@ -218,6 +189,55 @@ func ComparePodLogString(ctx context.Context, client klient.Client, customPod v1
...
+	}
+
+	var err error = nil
+	for retries > 0 {
+		podEventsDescriptions, podErr := getStringFromPod(ctx, client, pod, GetPodEventWarningDescriptions)
+		if podErr != nil {
+			return podErr
+		}
+
+		t.Logf("podEvents: %+v\n", podEventsDescriptions)
```
Comment (wainersm) [inline src/cloud-api-adaptor/test/e2e/assessment_helpers.go:237]: Hi @stevenhorsman !

What does the "+" in "+v" do? I vaguely remember I had to remove it to print the string correctly, I just don't remember where.

**Review by mkulke — APPROVED** [review]

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go (hunk 1 above)
Comment (stevenhorsman) [reply src/cloud-api-adaptor/test/e2e/assessment_helpers.go:237]: The `%+v` includes the field names of the struct. In this case it's a string IIRC, so I should just switch to `%s` instead. I think the `%+v` was from an earlier version. Sorry about that.

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go [hunk 2]
```diff
@@ -218,6 +189,55 @@ func ComparePodLogString(ctx context.Context, client klient.Client, customPod v1
...
+		if !strings.Contains(podEventsDescriptions, expectedPodEventDescribe) {
+			err = fmt.Errorf("error: Pod Descriptions don't contain Expected String %s", expectedPodEventDescribe)
```
[the remark this answers is not in the document]
Comment (stevenhorsman) [reply src/cloud-api-adaptor/test/e2e/assessment_helpers.go:239]: Thanks - I usually think of the pod events in the context of `kubectl describe`, so my inconsistencies leak in.

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go [hunk 3]
```diff
@@ -148,41 +119,49 @@ func WatchImagePullTime(ctx context.Context, client klient.Client, caaPod v1.Pod
...
+	caaPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "confidential-containers-system",
+		},
+	}
+	pods, err := GetPodNamesByLabel(ctx, client, t, caaPod.GetObjectMeta().GetNamespace(), "app", "cloud-api-adaptor")
+	if err != nil {
+		return nil, fmt.Errorf("getCaaPod: GetPodNamesByLabel failed: %v", err)
+	}
+	caaPod.Name = pods.Items[0].Name
```
[the remark this answers is not in the document]
Comment (stevenhorsman) [reply src/cloud-api-adaptor/test/e2e/assessment_helpers.go:132]: Yes definitely. I will add that in. Thanks for the suggestion.

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go [hunk 4]
```diff
@@ -218,6 +189,55 @@ func ComparePodLogString(ctx context.Context, client klient.Client, customPod v1
...
+	var descriptionsBuilder strings.Builder
+	for _, event := range events.Items {
+		if event.Type == v1.EventTypeWarning {
+			descriptionsBuilder.WriteString(event.Message)
+		}
+	}
+	return descriptionsBuilder.String(), nil
+}
+
+func ComparePodEventWarningDescriptions(ctx context.Context, t *testing.T, client klient.Client, pod v1.Pod, expectedPodEvent string) error {
```
Comment (wainersm) [inline src/cloud-api-adaptor/test/e2e/assessment_helpers.go]: Suggestion: document the function so users understand the logic easily. You did it very well on the commit message, btw.

File: src/cloud-api-adaptor/test/e2e/assessment_helpers.go [hunk 5]
```diff
@@ -124,33 +124,31 @@ func WatchImagePullTime(ctx context.Context, client klient.Client, caaPod v1.Pod
...
+	if err != nil {
 		t.Fatal(err)
 	}
-	for _, pod := range podlist.Items {
-		if pod.Labels["app"] == "cloud-api-adaptor" && pod.Spec.NodeName == nodeName {
-			podLogString, err := GetPodLog(ctx, client, pod)
-			if err != nil {
-				return false, err
-			}
+	caaPod.Name = pods.Items[0].Name
```
[the remark this answers is not in the document]
Comment (stevenhorsman) [reply src/cloud-api-adaptor/test/e2e/assessment_helpers.go]: That's an interesting point. I think we need to pass the nodeName into the CAA pod so we can check we are getting the correct one. Let me try that

[the remark this answers is not in the document]
Comment (wainersm) [reply src/cloud-api-adaptor/test/e2e/assessment_helpers.go]: I think it does address. thanks @stevenhorsman !

**Review by wainersm — APPROVED** [review]
Hey @stevenhorsman ! Great refactoring, the code is much nicer to read indeed!

**Pull request merged**